### PR TITLE
feat(auth): Add android samples for TOTP

### PR DESCRIPTION
**Pull request by sdhuka**
#### Description of changes:
* Add TOTP Sample for Android

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [X] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [ ] iOS
- [X] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [X] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [X] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://link.com)` 
            HTML: `<a href="https://link.com">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

**Comment by katiegoines** [comment]
@sdhuka - Seeing some merge conflicts that will block this PR from being merged. When you get a chance, please resolve these. Thanks!

**Comment by sdhuka** [comment]
> 

@gpanshu fyi

File: src/directory/directory.mjs [hunk 1]
```diff
@@ -180,12 +180,12 @@ export const directory = {
           {
             title: 'Working with files and attachments',
             route: '/lib/graphqlapi/working-with-files',
-            filters: ['js', 'ios']
+            filters: ['js', 'ios', 'android']
```
Comment (dnys1) [inline src/directory/directory.mjs:183]: ?

File: src/directory/directory.mjs [hunk 2]
```diff
@@ -330,7 +330,7 @@ export const directory = {
           {
             title: 'SMS flows',
             route: '/lib/auth/sms_flows',
-            filters: ['flutter', 'ios']
+            filters: ['flutter', 'ios', 'android']
```
Comment (dnys1) [inline src/directory/directory.mjs:333]: You need the one above this too (MFA)

Comment (dnys1) [reply src/directory/directory.mjs:333]: The SMS flows page requires code snippets. Do you want to add that in a separate PR or this one?

File: src/directory/directory.mjs [hunk 3]
```diff
@@ -215,7 +215,7 @@ export const directory = {
           {
             title: 'Upgrade guide from AppSync SDK',
             route: '/lib/graphqlapi/upgrade-guide',
-            filters: ['js', 'ios']
+            filters: ['js', 'ios', 'android']
```
Comment (dnys1) [inline src/directory/directory.mjs:218]: ?

File: src/directory/directory.mjs [hunk 4]
```diff
@@ -180,12 +180,12 @@ export const directory = {
...
             title: 'Working with files and attachments',
             route: '/lib/graphqlapi/working-with-files',
-            filters: ['js', 'ios']
+            filters: ['js', 'ios', 'android']
           },
           {
             title: 'Optimistic UI',
             route: '/lib/graphqlapi/optimistic-ui',
-            filters: ['js', 'ios']
+            filters: ['js', 'ios', 'android']
```
Comment (dnys1) [inline src/directory/directory.mjs:188]: ?

**Review by dnys1 — COMMENTED** [review]
You'll need changes to the sign-in steps page as well

File: src/directory/directory.mjs (hunk 2 above)
Comment (gpanshu) [reply src/directory/directory.mjs:328]: done

File: src/fragments/lib/auth/android/sms/update_mfa_preference.mdx [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+            "AuthQuickstart",
+            "Failed to update MFA preference",
+            e
+        )
+    }
+}
+```
+
+</Block>
+<Block name="RxJava">
```
Comment (mattcreaser) [inline src/fragments/lib/auth/android/sms/update_mfa_preference.mdx:42]: No RxJava API

File: src/fragments/lib/auth/android/sms/sign_in.mdx [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+```
+
+</Block>
+<Block name="RxJava">
+
+```java
+RxAmplify.Auth.signIn("username", "password")
+    .subscribe(
+        result -> {
+          if (result.nextStep.signInStep == AuthSignInStep.CONTINUE_SIGN_IN_WITH_TOTP_SETUP) {
```
Comment (mattcreaser) [inline src/fragments/lib/auth/android/sms/sign_in.mdx:67]: Kotlin syntax

File: src/fragments/lib/auth/android/sms/sign_in.mdx [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+```
+
+</Block>
+<Block name="Kotlin - Coroutines">
+
+```kotlin
+try {
+    val result = Amplify.Auth.signIn("username", "password")
+    if (result.nextStep.signInStep == AuthSignInStep.CONTINUE_SIGN_IN_WITH_TOTP_SETUP) {
+        val destination = result.nextStep.codeDeliveryDetails.destination
```
Comment (mattcreaser) [inline src/fragments/lib/auth/android/sms/sign_in.mdx:48]: `codeDeliveryDetails` is nullable

File: src/fragments/lib/auth/android/sms/sign_in.mdx [hunk 8]
```diff
@@ -0,0 +1,81 @@
+<BlockSwitcher>
+<Block name="Java">
+
+```java
+Amplify.Auth.signIn(
+    "username",
+    "password",
+    result -> {
+      if (result.nextStep.getSignInStep() == AuthSignInStep.CONFIRM_SIGN_IN_WITH_SMS_MFA_CODE) {
```
Comment (mattcreaser) [inline src/fragments/lib/auth/android/sms/sign_in.mdx:9]: Kotlin syntax

File: src/fragments/lib/auth/android/sms/sign_in.mdx [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+      String destination = result.nextStep.getCodeDeliveryDetails().getDestination();
+        Log.d("SignIn", "SMS code sent to "+ destination);
+        Log.d("SignIn", "Additional Info" + result.nextStep.getAdditionalInfo());
+
+        // Prompt the user to enter the SMSMFA code they received
+        // Then invoke `confirmSignIn` api with the code
+      }
+    },
+    error -> Log.e("AuthQuickstart", error.toString());
```
Comment (mattcreaser) [inline src/fragments/lib/auth/android/sms/sign_in.mdx:18]: Extra semi-colon

File: src/fragments/lib/auth/android/signin_next_steps/91_confirm_totp_mfa.mdx [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+} catch (error: Exception) {
+    Log.e("AuthQuickstart", "Unexpected error: $error")
+}
+```
+</Block>
+
+<Block name="Kotlin - Coroutines">
+
+```kotlin
+try {
```
Comment (mattcreaser) [inline src/fragments/lib/auth/android/signin_next_steps/91_confirm_totp_mfa.mdx:57]: Don't need try/catch here

File: src/fragments/lib/auth/android/signin_next_steps/91_confirm_totp_mfa.mdx [hunk 11]
```diff
@@ -0,0 +1,97 @@
+If the next step is `CONFIRM_SIGN_IN_WITH_TOTP_CODE`, Amplify Auth is waiting for the user to enter their TOTP code. To handle this step, your app's UI must prompt the user to enter the code. After the user enters the code, your implementation must pass the value to Amplify Auth `confirmSignIn` API.
+
+Note: the signIn result also includes an `AuthCodeDeliveryDetails` member. It includes additional information about the code delivery.
+
+<BlockSwitcher>
+
+<Block name="Java">
+
+```java
+try {
```
Comment (mattcreaser) [inline src/fragments/lib/auth/android/signin_next_steps/91_confirm_totp_mfa.mdx:10]: Don't need try/catch here

File: src/fragments/lib/auth/android/mfa_preference/20_update_mfa_preference.mdx [hunk 12]
```diff
@@ -0,0 +1,20 @@
+<BlockSwitcher>
+
+<Block name="Kotlin">
```
Comment (mattcreaser) [inline src/fragments/lib/auth/android/mfa_preference/20_update_mfa_preference.mdx:3]: Missing Java

File: src/fragments/lib/auth/android/mfa_preference/10_fetch_mfa_preference.mdx [hunk 13]
```diff
@@ -0,0 +1,14 @@
+<BlockSwitcher>
+
+<Block name="Kotlin">
+
+```kotlin
```
Comment (mattcreaser) [inline src/fragments/lib/auth/android/mfa_preference/10_fetch_mfa_preference.mdx:5]: Missing Java

**Review by mattcreaser — CHANGES_REQUESTED** [review]
There are a lot of syntax errors in the code snippets, intermixing of language features from Java and Kotlin, etc.

I've noted many of them in the following comments. Sometimes I only added one comment that applied to multiple places. We need to go through each example and make sure the code provided will actually compile.

**Pull request closed without merging**